### DISCUSSION TITLE
power on on boot

### DISCUSSION THREAD

**Discussion by montyfert** [arendst/Tasmota#18187] [Q&A]
hi am i doing something wrong ,
Rule1 ON Power1#boot DO POWER1 1 ENDON
RULE1 1

any help would be great thanks
martin

**Comment by barbudor** [comment]
Your rule looks correct
I would assume that as you try command "POWER1"' you have a relay or a light configured ?
Because the trigger won't work if you don't have a "power" to control

May be provide more details on your configuration (device, gpios, ...)

**Reply by sfromis** [reply]
Also activate `Weblog 3` for details, and check the console right after boot.

Anyway, the intended feature for booting with power on is to use the command `PowerOnState` to select that.

**Comment by montyfert** [comment]
hi im using it with a TM1637 to display info from my 3d printer.
![image](https://user-images.githubusercontent.com/74438819/224866019-d91020c2-8a7c-4c32-b195-aea85eee42cf.png)
![image](https://user-images.githubusercontent.com/74438819/224866720-e71810e7-8f5b-4715-a7ca-06b50cf6a33e.png)

**Reply by barbudor** [reply]
Ah, so the Power1 is not a relay or a light but the 7-segment display
That may be a not handled case

At least You can work around by using a 'system#boot' or 'system#init' trigger

**Comment by montyfert** [comment]
thanks  if i do 
Rule1 ON system#boot DO POWER1 1 ENDON 
rule1 1
restart i get 
08:38:30.161 MQT: stat/mini_display_2/RESULT = {"POWER1":"ON"}
08:38:30.166 MQT: stat/mini_display_2/POWER1 = ON
08:38:30.178 RUL: SYSTEM#BOOT performs "POWER1 1"
08:38:30.185 MQT: stat/mini_display_2/RESULT = {"POWER1":"ON"}
08:38:30.188 MQT: stat/mini_display_2/POWER1 = ON
08:38:30.379 MQT: stat/mini_display_2/RESULT = {"DisplayText":"100% "}
08:38:30.426 MQT: stat/mini_display_2/RESULT = {"Command":"Unknown"}
08:38:30.475 MQT: stat/mini_display_2/RESULT = {"Command":"Unknown"}
08:38:30.527 MQT: stat/mini_display_2/RESULT = {"POWER1":"OFF"}
08:38:30.530 MQT: stat/mini_display_2/POWER1 = OFF
08:38:30.585 MQT: stat/mini_display_2/RESULT = {"Command":"Unknown"}
08:38:32.439 QPC: Reset

but if i do 
Rule1 ON system#boot DO POWER1 0 ENDON 
rule1 1
i get 
08:42:37.187 MQT: stat/mini_display_2/RESULT = {"POWER1":"OFF"}
08:42:37.191 MQT: stat/mini_display_2/POWER1 = OFF
08:42:37.204 RUL: SYSTEM#BOOT performs "POWER1 0"
08:42:37.210 MQT: stat/mini_display_2/RESULT = {"POWER1":"OFF"}
08:42:37.212 MQT: stat/mini_display_2/POWER1 = OFF
08:42:37.403 MQT: stat/mini_display_2/RESULT = {"DisplayText":"100% "}
08:42:37.452 MQT: stat/mini_display_2/RESULT = {"Command":"Unknown"}
08:42:37.503 MQT: stat/mini_display_2/RESULT = {"Command":"Unknown"}
08:42:37.553 MQT: stat/mini_display_2/RESULT = {"POWER1":"ON"}
08:42:37.556 MQT: stat/mini_display_2/POWER1 = ON
08:42:37.604 MQT: stat/mini_display_2/RESULT = {"Command":"Unknown"}
08:42:39.514 QPC: Reset

thanks 
martin

**Reply by sfromis** [reply]
Maybe activate `Weblog 3` to debug and fix those `{"Command":"Unknown"}`

That would likely also show you the source of the 2nd power on/off you see.
